lib/Dialect/FIRRTL/Ops.cpp

```diff
@@ -455,11 +455,11 @@ static ParseResult parseFExtModuleOp(OpAsmParser &parser,
   return parseFModuleOp(parser, result, /*isExtModule:*/ true);
 }
 
-static LogicalResult verifyFModuleOp(FModuleOp module) {
+static LogicalResult verifyFModuleOp(FModuleOp &module) {
   // The parent op must be a circuit op.
-  auto *parentOp = module.getParentOp();
-  if (!parentOp || !isa<CircuitOp>(parentOp)) {
-    module.emitOpError("should be embedded into a firrtl.circuit");
+  auto parentOp = dyn_cast<CircuitOp>(module.getParentOp());
```

Comment:
I think this is ok, but random FYI: "dyn_cast" does not allow a null operand.  Use "dyn_cast_or_null" if you need that.  That said, in this case, I think the verify hook only gets called when things are in a container, so this should be ok.

Reply:
Thanks for the info! I'll go with `dyn_cast_or_null`.